pkg/sql/sqlbase/table.go

```diff
@@ -1584,7 +1585,24 @@ func EncodeInvertedIndexKeys(
 func EncodeInvertedIndexTableKeys(val tree.Datum, inKey []byte) (key [][]byte, err error) {
 	switch t := tree.UnwrapDatum(nil, val).(type) {
 	case *tree.DJSON:
-		return (t.JSON).EncodeInvertedIndexKeys(inKey)
+		keys, err := (t.JSON).EncodeInvertedIndexKeys(inKey)
+		if err != nil {
+			return nil, err
+		}
+		// unsafe is safe to use here because we're not modifying the slice.
```

Comment:
Add a comment here that you're getting rid of duplicate keys

Reply:
Is it possible to convert EncodeInvertedIndexKeys() to not return duplicates in the first place?

Reply:
Converting it to not return duplicates in the first place is what the discussion about k-way merge is about.